docs-client/src/components/GotoSelect/index.tsx

```diff
@@ -105,10 +105,17 @@ const GotoSelect: React.FunctionComponent<GotoSelectProps> = ({
 }) => {
   const classes = useStyles();
 
-  const handleSelection = useCallback(
-    (_: ChangeEvent<{}>, option: Option | null): void => {
-      if (option) {
-        navigateTo(option.value);
+  const value = useRef('');
```

Comment:
I tried using `useState()`, but that triggers a component re-render, which we don't want.
`let value = '';` would also work, but I think `useRef('')` is preferred. I'm not very familiar with React.

Reply:
If we want to keep the value between renders, we should use `useRef()`, but otherwise `let` is OK. https://stackoverflow.com/questions/56553236/differences-between-useref-and-ref-variable-in-reactjs

Reply:
I'll do some more tests, but I think the value should not persist between renders. I assumed `useRef()` will make a new reference when the whole `GotoSelect` component is rendered.

The component is rendered only once either way, so maybe both are okay. I'll look into it in more detail and decide which approach to use.

Reply:
I played around by forcing renders and it looks like the `Autocomplete` component retains the value.
So I decided to keep `useRef()` so we can track the value between renders.

I also memoized `options` so we don't have to call `getOptions()` on every render, which also gets rid of a warning.
```
Material-UI: The value provided to Autocomplete is invalid.
None of the options match with `{"group":"Services","label":"example.springframework.boot.minimal.HelloAnnotatedService#defaultHello|GET","value":"/methods/example.springframework.boot.minimal.HelloAnnotatedService/defaultHello/GET"}`.
You can use the `getOptionSelected` prop to customize the equality test.
```

I'll do more tests on an internal server tomorrow, to verify if everything works as expected.

Reply:
Thanks, @KarboniteKream !